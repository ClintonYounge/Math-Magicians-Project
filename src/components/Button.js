// src/components/Button.js

import React from 'react';
import PropTypes from 'prop-types';

const Button = ({
  name, clickHandler, orange = false, wide = false,
}) => {
  const handleClick = () => {
    clickHandler(name);
  };

  const className = [
    'component-button',
    orange ? 'orange' : '',
    wide ? 'wide' : '',
  ].join(' ').trim();

  return (
    <div className={className}>
      <button type="button" onClick={handleClick}>{name}</button>
    </div>
  );
};

Button.defaultProps = {
  orange: false,
  wide: false,
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
  orange: PropTypes.bool,
  wide: PropTypes.bool,
};

export default Button;
