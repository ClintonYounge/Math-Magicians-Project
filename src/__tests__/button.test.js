import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from '../components/Button';

test('renders button with correct name', () => {
  const buttonName = 'AC';
  const { getByText } = render(<Button name={buttonName} clickHandler={() => {}} />);
  const button = getByText(buttonName);
  expect(button).toBeInTheDocument();
});

test('calls click handler when button is clicked', () => {
  const buttonName = 'AC';
  const clickHandler = jest.fn();
  const { getByText } = render(<Button name={buttonName} clickHandler={clickHandler} />);
  const button = getByText(buttonName);
  fireEvent.click(button);
  expect(clickHandler).toHaveBeenCalledTimes(1);
});
