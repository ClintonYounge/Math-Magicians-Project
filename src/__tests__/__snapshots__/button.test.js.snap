// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button with correct name: 
<button
  type="button"
>
  AC
</button>
 1`] = `
<button
  type="button"
>
  AC
</button>
`;
